src/Adapter/Form/ChoiceProvider/TaxRuleGroupChoiceProvider.php

```diff
@@ -26,26 +26,48 @@
 
 namespace PrestaShop\PrestaShop\Adapter\Form\ChoiceProvider;
 
+use PrestaShop\PrestaShop\Core\Form\FormChoiceAttributeProviderInterface;
 use PrestaShop\PrestaShop\Core\Form\FormChoiceProviderInterface;
 use TaxRulesGroup;
 
 /**
  * Provides tax rule group choices with tax rule name as key and id as value
  */
-final class TaxRuleGroupChoiceProvider implements FormChoiceProviderInterface
+final class TaxRuleGroupChoiceProvider implements FormChoiceProviderInterface, FormChoiceAttributeProviderInterface
 {
     /**
      * {@inheritdoc}
      */
     public function getChoices()
     {
-        $rules = TaxRulesGroup::getTaxRulesGroupsForOptions();
-
         $choices = [];
-        foreach ($rules as $rule) {
+        foreach ($this->getRules() as $rule) {
             $choices[$rule['name']] = $rule['id_tax_rules_group'];
         }
 
         return $choices;
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getChoicesAttributes()
```

Comment:
Missing return type

Reply:
I thought it would conflict with the interface prototypes that don't have strict typing, but it seems to be fine!